src/main/content/_assets/js/javadoc-search.js

```diff
@@ -358,26 +360,26 @@ $(function() {
                     // calling replace so as not to add an entry to the history
                     parent.classFrame.location.replace(pathtoroot + url);
 
-                    // building the hash
-                    var hash = top.window.location.hash;
-                    if (hash.indexOf("class=") !== -1) {
+                    // building the query parameters
+                    var newURL = new URL(top.window.location.href);
+                    var queryParams = newURL.searchParams;
+                    
+                    if (queryParams.has('class')) {
                         try {
-                            var hashNameToMatch = "(.*)class=.*?.html(.*)";
-                            var regExpToMatch = new RegExp(hashNameToMatch, "g");
-                            var groups = regExpToMatch.exec(hash);
-                            hash = groups[1] + "class=" + url + groups[2];
+                            queryParams.set('class', url);
                         } catch (ex) {
 
                         }
                     } else {
-                        hash = "#class=" + url + "&package=allclasses-frame.html"; // putting in default html for package iframe
-                    }
-                    if (hash !== top.window.location.hash) {
-                        var state = {};
-                        state["iframe.rightIframe"] = href;
-                        // create a history entry
-                        top.window.history.pushState(state, null, hash);
+                        queryParams.set('class', url);
+                        queryParams.set('package', 'allclasses-frame.html');
```

Comment:
Can you give me some context on what is happening here?  To my untrained eye, it looks like we are calling `queryParams.set('class', url);` regardless of whether `queryParams.has('class')` is `true` or `false`.

Reply:
I was just changing the logic from hash to QP. I do see why we don't need to set it in both cases, but i'll figure it out today and improve it.